In [1]:
# Sempre a primeira célula do notebook, importar o que será usado
# Se aparecer erro, a biblioteca não está instalada
# Solução: Abrir o terminar e roda. pip install pandas openpyxl


import pandas as pd  # Biblioteca de análise de dados
import re  # Expressão regulares (para limpar textos)
from datetime import datetime  # Biblioteca que trabalha com datas


# Confirmar que as importações funcionaram

print(" pandas versão:", pd.__version__)
print(" Bibliotecas importadas com sucesso!")

 pandas versão: 3.0.2
 Bibliotecas importadas com sucesso!


In [ ]:
# pd.read_csv()         # para ler arquivos .csv
# CAMINHO = onde está o arquivo. Como ele está na MESMA pasta no notebook.......??????
# basta passar o nome do arquivo , entre aspas, para função ler o arquivo e criar um DataFrame (tabela) com os dados.


CAMINHO = CAMINHO = (
    r"C:\Users\waldinei.rosa\OneDrive\Área de Trabalho\CURSOPYTHON2026\ANALISE_DADOS_SCTEC\AULA_GIT\aulas_semana_04\turma-visualizacao-de-dados\alunos\waldinei_lameira\aula_01\base_rh.csv"
)

df = pd.read_csv(
    CAMINHO,
    sep=";",  # separador de colunas é ;. Precisa verificar na base do arqulivo xls
    encoding="cp1252",  # encoding do windows, serve para ler acentos
    decimal=",",  # virgula como separador decimal
)

print("Dados carregador com sucesso!")
print(f" Total de linhas: {df.shape[0]}")
print(f" Total de colunas: {df.shape[1]}")


Dados carregador com sucesso!
 Total de linhas: 1000
 Total de colunas: 10


In [5]:
# df.head(n)                # para ver as primeiras linhas da tabela, e conferir se os dados foram lidos corretamente
# Padrão: df.head(5)        # mostra as 5 primeiras linhas, mas podem passar  outro número para ver mais ou menos linhas. Se não passar nada o padrão é 5.
# Use essas informações para ter uma visão rápida dos dados, e confirma que as colunas foram lidas corretamente, e que os dados estão no formato esperado.

df.head(5)


,ID_Funcionario,Nome,Departamento,Cargo,Salario,Data_Admissao,Genero,Idade,Estado_Civil,Status
0,1,Julia Nunes,Logística,Coordenador,9088.34,13/08/2024,F,43,Solteiro,Inativo
1,2,Sr. Gustavo Duarte,TI,Gerente,8155.98,29/04/2017,F,59,Solteiro,Inativo
2,3,Srta. Mariana Cunha,RH,Coordenador,14027.93,11/12/2024,F,27,Divorciado,Ativo
3,4,Ana Sophia da Cruz,Produção,Técnico,6647.97,16/06/2019,M,50,Casado,Inativo
4,5,Dr. Nicolas Pinto,Produção,Gerente,8730.19,29/03/2019,F,62,Casado,Inativo


In [6]:
# df.tail(n)            # para ver as últimas linhas do tabela, e confirmar que os dados foram lidos corretamente
# Serve para confirmar que o final da tabela também foi lida corretamente, e verifica se tem alguma informação importante (ex. rodapés,  notas, etc..,)
# Se o arquivo tiver 1000 linhas, o tail deve mostar os IDs (próximos de 1000) de 996 a 100, confirmando que todas as linhas foram lidas

df.tail(10)

,ID_Funcionario,Nome,Departamento,Cargo,Salario,Data_Admissao,Genero,Idade,Estado_Civil,Status
990,991,Sr. Vicente Silveira,Produção,Gerente,8026.70,11/07/2021,F,55,Viúvo,Inativo
991,992,Emanuel da Cruz,Logística,Gerente,12572.11,30/01/2024,M,33,Viúvo,Ativo
992,993,Enrico Monteiro,RH,Coordenador,14677.33,26/12/2017,M,52,Divorciado,Inativo
993,994,Ana Lívia Cardoso,RH,Coordenador,14912.48,22/10/2015,M,27,Solteiro,Inativo
994,995,Fernando Monteiro,RH,Assistente,10983.02,09/01/2023,F,42,Viúvo,Ativo
995,996,Matheus Aragão,Financeiro,Analista,11389.03,06/05/2022,F,26,Viúvo,Ativo
996,997,Ryan Martins,Financeiro,Analista,14060.42,05/06/2023,F,37,Viúvo,Ativo
997,998,Isabelly Ferreira,TI,Gerente,2348.73,29/12/2023,F,58,Solteiro,Inativo
998,999,Thales Farias,Produção,Analista,9430.62,25/12/2019,M,41,Solteiro,Ativo
999,1000,Sra. Maysa Costela,Produção,Técnico,12549.72,14/11/2022,M,46,Viúvo,Inativo


In [ ]:
# df.columns            # para ver o nome das colunas, e confirmar que foram lidas corretamente, e que estão no formato esperado(ex: sem caracteres estranhos, sem espaços, etc...)
# tolist()              # para tranformar o index de colunas em uma lista, e facilita a manipulação dos nomes das colunas (ex: para renomear, para acessar as colunas, etc...)

print("Colunas do dataset: ")
print(df.columns.tolist())

Colunas do dataset: 
['ID_Funcionario', 'Nome', 'Departamento', 'Cargo', 'Salario', 'Data_Admissao', 'Genero', 'Idade', 'Estado_Civil', 'Status']


## Diagnóstico: tipos e nulos

**O que são tipos de dados?**
Cada coluna armazena um tipo de informação. O Python precisa saber o tipo para saber o que pode fazer com ele.

| Tipo | O que é | Exemplo |
|------|---------|---------|
| `int64` | Número inteiro | 1, 42, 1900 |
| `float64` | Número decimal | 9088.34, 3.14 |
| `object` | Texto / string | "Julia Nunes", "Abvo" |
| `datetime64` | Data e hora | 2024-08-13 |

**O que são nulos?**
São células vazias — o campo não foi preenchido no sistema de origem.
Em Python/pandas, nulos aparecem como `NaN` (Not a Number) ou `NaT` (Not a Time).

In [8]:
## df.info()
# Nome de cada coluna
# Quantidade de valores não-nulos
# Tipo do dado (dtype)
# Uso de memória

# É o comando mais importante para o primeiro diagnóstico!

df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   ID_Funcionario  1000 non-null   int64  
 1   Nome            1000 non-null   str    
 2   Departamento    1000 non-null   str    
 3   Cargo           1000 non-null   str    
 4   Salario         1000 non-null   float64
 5   Data_Admissao   1000 non-null   str    
 6   Genero          1000 non-null   str    
 7   Idade           1000 non-null   int64  
 8   Estado_Civil    1000 non-null   str    
 9   Status          1000 non-null   str    
dtypes: float64(1), int64(2), str(7)
memory usage: 78.3 KB


In [ ]:
# df.snull() cria uma tabela de True/False (True= é nulo)
# sum() soma os True de cada coluna(True vale 1, False vale 0)
# Resultado: quantidade de nulos por coluna

print("Quantidade de nulos por coluna:")
print(df.isnull().sum())

print()

# Percentual de nulo - mais útil em grandes bases, para enterder a proporção de nulos em cada coluna.
# len(df) - qauntidade total de linhas do DataFrame

print("Percentual de nulos por coluna:")
print((df.isnull().sum() / len(df)) * 100)

Quantidade de nulos por coluna:
ID_Funcionario    0
Nome              0
Departamento      0
Cargo             0
Salario           0
Data_Admissao     0
Genero            0
Idade             0
Estado_Civil      0
Status            0
dtype: int64

Percentual de nulos por coluna:
ID_Funcionario    0.0
Nome              0.0
Departamento      0.0
Cargo             0.0
Salario           0.0
Data_Admissao     0.0
Genero            0.0
Idade             0.0
Estado_Civil      0.0
Status            0.0
dtype: float64


## Estatísticas das colunas numéricas

`describe()` gera um resumo estatístico automático.
Você vai usar isso para detectar valores suspeitos (ex: salário de R$0,00 ou idade de 200 anos).

| Estatística | O que significa |
|-------------|----------------|
| `count` | Quantos valores não-nulos existem |
| `mean` | Média |
| `std` | Desvio padrão — o quanto os valores variam em torno da média |
| `min` | Menor valor |
| `25%` | 25% dos registros estão abaixo desse valor |
| `50%` | Metade dos registros estão abaixo (mediana) |
| `75%` | 75% dos registros estão abaixo desse valor |
| `max` | Maior valor |

In [ ]:
df.describe().round(
    2
)  # para ver a estatistica descritiva das colunas numéricas, e confirmar que os dados estão no formato esperado (ex: sem valores estranhos, sem outliers, etc...)


,ID_Funcionario,Salario,Idade
count,1000.00,1000.00,1000.00
mean,500.50,8579.95,41.40
std,288.82,3657.37,13.67
min,1.00,2000.71,18.00
25%,250.75,5564.54,30.00
50%,500.50,8571.13,41.00
75%,750.25,11554.63,53.00
max,1000.00,14954.51,65.00


## CÉLULA 6 — Explorando colunas categóricas

**Coluna categórica** = coluna com poucos valores possíveis (departamento, cargo, status...)

- `nunique()` conta quantos valores diferentes existem
- `value_counts()` mostra a distribuição — quantas vezes cada valor aparece

In [ ]:
# df.nunique() - retorna quantos valores ÚNICOS cada coluna tem.
# serve para identificar qauis são categóricas (poucas opções)
# e quais são identificadoras (muitas opções, ex: ID, nome, etc...)

print("Valores únicos por coluna:")
print(df.nunique())


Valores únicos por coluna:
ID_Funcionario    1000
Nome               976
Departamento         6
Cargo                5
Salario           1000
Data_Admissao      878
Genero               2
Idade               48
Estado_Civil         4
Status               2
dtype: int64


In [ ]:
# df.['coluna'].value_counts() - para ver a quantidade de cada valor único em uma coluna específica, e entender a distribuição dos dados nessa coluna (ex: para identificar qual é o valor mais comum, se tem valores estranhos, etc...)
# vamos ver todas as colunas categóricas de uma vez, usando um loop for para iterar sobre as colunas, e verificar quais são categóricas (poucos valores únicos)
# 'for col in lista' = para cada coluna na lista, executa o bloco abaixo.

for col in ["Departamento", "Cargo", "Genero", "Status", "Estado_Civil"]:
    print(f"\n {col}")  # f-string: {} insere o valor da variável
    print(df[col].value_counts())



 Departamento
Departamento
Financeiro    189
Produção      182
RH            166
Vendas        160
Logística     156
TI            147
Name: count, dtype: int64

 Cargo
Cargo
Coordenador    225
Gerente        221
Assistente     199
Analista       178
Técnico        177
Name: count, dtype: int64

 Genero
Genero
F    510
M    490
Name: count, dtype: int64

 Status
Status
Ativo      515
Inativo    485
Name: count, dtype: int64

 Estado_Civil
Estado_Civil
Solteiro      265
Viúvo         261
Divorciado    240
Casado        234
Name: count, dtype: int64


## CÉLULA 7 — Selecionando colunas e filtrando linhas

São as operações mais usadas no dia a dia. Você vai usar isso em todo projeto.

- **Seleção de coluna** → `df['Nome']`
- **Seleção de múltiplas colunas** → `df[['Nome', 'Salario']]`
- **Filtro de linhas** → `df[df['Status'] == 'Ativo']`

In [ ]:
# seleciona uma coluna - retorna uma série (lista com índices)
# uma série é uma coluna de planilha isolada

df["Cargo"].head()

0    Coordenador
1        Gerente
2    Coordenador
3        Técnico
4        Gerente
Name: Cargo, dtype: str

In [ ]:
# Seleciona MÚLTIPLAS colunas - retorna um DataFrame (tabela)
# Nota os dois colchetes: df[['coluna1', 'coluna2', 'coluna3']]
# O primeiro [] é o operador de acesso ao DataFrame
# O segundo [] é a linha de colunas

df[["Nome", "Departamento", "Cargo", "Salario", "Genero"]].head(6)

,Nome,Departamento,Cargo,Salario,Genero
0,Julia Nunes,Logística,Coordenador,9088.34,F
1,Sr. Gustavo Duarte,TI,Gerente,8155.98,F
2,Srta. Mariana Cunha,RH,Coordenador,14027.93,F
3,Ana Sophia da Cruz,Produção,Técnico,6647.97,M
4,Dr. Nicolas Pinto,Produção,Gerente,8730.19,F
5,Marcela Costa,RH,Analista,12403.42,M


In [ ]:
# FILTRO SIMPLES: selecionar só as linhas onde status = 'Ativo'
# df['Status'] == 'Ativo'  → cria uma Series de True/False
# df[ True/False ]         → mantém só as linhas onde é True

ativos = df[df["Status"] == "Ativo"]

print(f"Total de funcionários: {len(df)}")
print(f"Funcionários ativos: {len(ativos)}")
print(f"Funcionários inativos: {len(df) - len(ativos)}")

Total de funcionários: 1000
Funcionários ativos: 515
Funcionários inativos: 485


In [ ]:
# FILTRO COMPOSTO: duas condições ao mesmo tempo
# Use & para AND (as duas condições precisam ser verdadeiras)
# Use | para OR (pelo menos uma precisa ser verdadeira)
# IMPORTANTE: cada condição precisa estar entre parênteses

gerentes_ti = df[(df["Cargo"] == "Gerente") & (df["Departamento"] == "TI")]

print(f"Gerentes na área de TI: {len(gerentes_ti)}")
gerentes_ti[["Nome", "Salario", "Status", "Data_Admissao"]]

Gerentes na área de TI: 31


,Nome,Salario,Status,Data_Admissao
1,Sr. Gustavo Duarte,8155.98,Inativo,29/04/2017
49,Benjamin da Costa,9766.01,Inativo,27/04/2024
130,Catarina Costela,9597.83,Inativo,03/07/2023
131,Laura Cardoso,7364.46,Inativo,29/06/2018
175,Maria Cardoso,12127.27,Inativo,22/05/2024
187,Ana Júlia Cardoso,8738.63,Ativo,30/05/2021
247,Ian Pereira,2395.32,Ativo,10/05/2019
273,Ana Clara Peixoto,7301.14,Ativo,16/11/2019
296,Augusto Santos,8237.80,Inativo,26/08/2021
307,Manuela Barbosa,7015.91,Ativo,21/06/2017


In [ ]:
# Estatistica rápida de uma coluna numéria
# .mean() - média
# .median() - mediana
# .min() - valor mínimo
# .max() - valor máximo

print(f"Salário médio: {df['Salario'].mean():,.2f}")
print(f"Salário mediano: {df['Salario'].median():,.2f}")
print(f"Salário mínimo: {df['Salario'].min():,.2f}")
print(f"Salário máximo: {df['Salario'].max():,.2f}")


Salário médio: 8,579.95
Salário mediano: 8,571.13
Salário mínimo: 2,000.71
Salário máximo: 14,954.51
